subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -323,24 +324,29 @@ private SnapshotHierarchy withWatcherChangeErrorHandling(SnapshotHierarchy curre
         try {
             return supplier.get();
         } catch (Exception ex) {
-            logWatchingError(ex, FILE_WATCHING_ERROR_MESSAGE_DURING_BUILD);
+            logWatchingError(ex, FILE_WATCHING_ERROR_MESSAGE_DURING_BUILD, null);
             return stopWatchingAndInvalidateHierarchyAfterError(currentRoot);
         }
     }
 
-    private void logWatchingError(Exception exception, String fileWatchingErrorMessage) {
-        if (exception instanceof InotifyInstanceLimitTooLowException) {
-            LOGGER.warn("{}. The inotify instance limit is too low. See {} for more details.",
-                fileWatchingErrorMessage,
-                daemonDocumentationIndex.getLinkToSection("sec:inotify_instances_limit")
-            );
-        } else if (exception instanceof InotifyWatchesLimitTooLowException) {
-            LOGGER.warn("{}. The inotify watches limit is too low.", fileWatchingErrorMessage);
-        } else if (exception instanceof WatchingNotSupportedException) {
-            // No stacktrace here, since this is a known shortcoming of our implementation
-            LOGGER.warn("{}. {}.", fileWatchingErrorMessage, exception.getMessage());
-        } else {
-            LOGGER.warn(fileWatchingErrorMessage, exception);
+    private void logWatchingError(Exception exception, String fileWatchingErrorMessage, @Nullable WatchMode watchMode) {
+        // Only log warnings when watching is explicitly enabled or on the info log.
+        // Since we don't know the watch mode during the build, we'll only log on info there.
+        boolean shouldLogWarnings = watchMode == WatchMode.ENABLED || LOGGER.isInfoEnabled();
```

Comment:
Why not pass `watchingExplicitlyEnabled` as a boolean instead to avoid the `@Nullable`?

Reply:
Yeah, I did that before, though then I would need to pass `false` in the places where I don't know the watch mode, so I preferred to pass in the WatchMode directly and not the boolean.